specification/azurestackhci/resource-manager/Microsoft.AzureStackHCI/preview/2023-07-01-preview/virtualMachineInstances.json

```diff
@@ -0,0 +1,1545 @@
+            },
+            "status": {
+              "type": "string",
+              "description": "The status of the operation performed on the virtual machine instance [Succeeded, Failed, InProgress]",
+              "enum": [
+                "Succeeded",
+                "Failed",
+                "InProgress"
+              ],
+              "x-ms-enum": {
+                "name": "Status",
+                "modelAsString": true
+              }
+            }
+          }
+        }
+      },
+      "description": "The observed state of virtual machine instances"
+    },
+    "VirtualMachineInstance": {
+      "type": "object",
+      "x-ms-client-flatten": true,
+      "x-ms-azure-resource": true,
+      "properties": {
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/VirtualMachineInstanceProperties"
+        },
+        "extendedLocation": {
+          "$ref": "./common.json#/definitions/ExtendedLocation",
+          "description": "The extendedLocation of the resource."
+        },
+        "identity": {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/Identity"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "../../../../../common-types/resource-management/v3/types.json#/definitions/ProxyResource"
+        }
+      ],
+      "description": "The virtual machine instance resource definition."
+    },
+    "VirtualMachineInstanceView": {
+      "type": "object",
+      "properties": {
+        "vmAgent": {
+          "$ref": "#/definitions/VirtualMachineVMConfigAgentInstanceView",
+          "description": "The VM Config Agent running on the virtual machine."
+        }
+      },
+      "description": "The instance view of a virtual machine."
+    },
+    "VirtualMachineVMConfigAgentInstanceView": {
+      "type": "object",
+      "properties": {
+        "vmVMConfigAgentVersion": {
+          "type": "string",
+          "description": "The VM Config Agent full version."
+        },
+        "statuses": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/InstanceViewStatus"
+          },
+          "x-ms-identifiers": [],
+          "description": "The resource status information."
+        }
+      },
+      "description": "The instance view of the VM Config Agent running on the virtual machine."
+    },
+    "InstanceViewStatus": {
+      "type": "object",
+      "properties": {
+        "code": {
+          "type": "string",
+          "description": "The status code."
+        },
+        "level": {
+          "type": "string",
+          "description": "The level code.",
+          "enum": [
+            "Info",
+            "Warning",
+            "Error"
+          ],
+          "x-ms-enum": {
+            "name": "StatusLevelTypes",
+            "modelAsString": true
+          }
+        },
+        "displayStatus": {
```

Comment:
>displayStatus [](http://example.com/codeflow?start=9&length=13)

can it be defined as enum?

Reply:
@pathuang-sia What do you think? Do you know which code generates this `displayStatus`?